lib/chef/resource/msu_package.rb

```diff
@@ -24,23 +24,18 @@ class Resource
     class MsuPackage < Chef::Resource::Package
       include Chef::Mixin::Uris
 
+      resource_name :msu_package
       provides :msu_package, os: "windows"
 
       allowed_actions :install, :remove
-
-      def initialize(name, run_context = nil)
-        super
-        @resource_name = :msu_package
-        @source = name
-        @action = :install
-      end
+      default_action :install
 
       property :source, String,
                 coerce: (proc do |s|
                   unless s.nil?
                     uri_scheme?(s) ? s : Chef::Util::PathHelper.canonical_path(s, false)
                   end
-                end)
+                end), name_property: true
```

Comment:
Might want to put this before the `coerce` so it's easier to see when scanning.

Reply:
Fixed